registry/session_v2.go

```diff
@@ -124,14 +124,15 @@ func (r *Session) HeadV2ImageBlob(ep *Endpoint, imageName, sumType, sum string,
 		return false, err
 	}
 	res.Body.Close() // close early, since we're not needing a body on this call .. yet?
-	switch res.StatusCode {
-	case 200:
+	switch {
+	case res.StatusCode >= 200 && res.StatusCode < 400:
```

Comment:
It's simply that a 301 is not an error. If a 2xx or 3xx response is issued, it should be assumed that the content is known to the registry and *does not need to be pushed*. Whether or not its actually available at that location doesn't matter. While the registry may never return this error code, the client should assume that it's a success and continue.

In reality, 301 should be handled transparently by the Go http client, resulting in an eventual 200 or 404 response. However, opening up the accepted error codes here allows us to modify the protocol and maintain backwards compatibility with 1.5.